x/inflation/types/v1beta2/inflation_calculator.go

```diff
@@ -17,74 +18,85 @@ func GetInflationCalculator(
-		inflationRateChange := inflationRateChangePerYear.Quo(sdk.NewDec(int64(params.BlocksPerYear)))
+	// years passed since genesis = seconds passed since genesis / number of seconds per year
+	// can be a fraction, eg: 0.5
+	yearsPassed := decimal.WithPrecision(sdk.Precision).
+		Quo(
+			// seconds since genesis
+			decimal.WithPrecision(sdk.Precision).
+				Sub(
+					decimal.New(btime.Unix(), 0),
+					decimal.New(gtime.Unix(), 0),
+				),
+			// Number of hours in an year = 8766 (averaging the leap year hours)
+			// Number of minutes in an hour = 60
+			// Number of seconds in a minute = 60
+			// => Number of seconds per year = 60 * 60 * 8766 = 31557600
+			decimal.New(31557600, 0),
+		)
+	// 2^(-t/tHalf)
+	pow := decimal.WithPrecision(sdk.Precision)
+	pow = pow.Context.
+		Pow(
+			pow,
+			decimal.New(2, 0),
+			decimal.WithPrecision(sdk.Precision).
+				Mul(
+					decimal.New(-1, 0),
+					decimal.WithPrecision(sdk.Precision).
+						Quo(yearsPassed, inflationDecayFactor),
+				),
+		)
+	// convert pow to sdk.Dec with a 6 unit precision: sdk.Decimal(big.Int(pow * 10^6)) / 10^6
+	powDec := sdk.NewDecFromBigInt(
+		decimal.WithPrecision(sdk.Precision).
+			Mul(pow, decimal.New(1000000, 0)).
+			Int(nil),
+	).QuoInt64(1000000)
+
+	// yearsPassed :=
+	idealInflation := iparams.InitialInflation.Mul(powDec)
 
-		// note inflationRateChange may be negative
-		currentInflation := minter.Inflation.Add(inflationRateChange)
+	// (1 - bondedRatio/GoalBonded) * InflationRateChange
+	inflationRateChangePerYear := sdk.OneDec().
+		Sub(bondedRatio.Quo(mparams.GoalBonded)).
+		Mul(mparams.InflationRateChange)
+	inflationRateChange := inflationRateChangePerYear.Quo(sdk.NewDec(int64(mparams.BlocksPerYear)))
 
-		// min, max currentInflation based on a defined range parameter 'r'
-		// currentInflation range = [I(t) - I(t) * R, I(t) + I(t) * R]
-		r := inflationParams.Variance
-		minInflation := idealInflation.Sub(idealInflation.Mul(r))
-		maxInflation := idealInflation.Add(idealInflation.Mul(r))
+	// note inflationRateChange may be negative
+	currentInflation := minter.Inflation.Add(inflationRateChange)
 
-		// minInflation >= minimumMinInflation
-		minimumMinInflation := sdk.ZeroDec() // 0 for now
-		if minInflation.LT(minimumMinInflation) {
-			minInflation = minimumMinInflation
-		}
+	// min, max currentInflation based on a defined range parameter 'r'
+	// currentInflation range = [I(t) - I(t) * R, I(t) + I(t) * R]
+	r := iparams.Variance
+	minInflation := idealInflation.Sub(idealInflation.Mul(r))
+	maxInflation := idealInflation.Add(idealInflation.Mul(r))
 
-		if currentInflation.LT(minInflation) {
-			currentInflation = minInflation
-		} else if currentInflation.GT(maxInflation) {
-			currentInflation = maxInflation
-		}
+	// minInflation >= minimumMinInflation
+	minimumMinInflation := sdk.ZeroDec() // 0 for now
+	if minInflation.LT(minimumMinInflation) {
+		minInflation = minimumMinInflation
+	}
 
-		return currentInflation
```

Comment:
I think this can just be written as 
```
currentInflation = sdk.MaxDec(currentInflation, minInflation)
currentInflation = skd.MinDec(currentInflation, maxInflation)
```